From 258409cfa05aaa378e4f120dc67fb226465fa829 Mon Sep 17 00:00:00 2001
From: Runar Borge <runar@borge.nu>
Date: Sat, 23 Jan 2021 00:15:41 +0100
Subject: [PATCH] frr-reload: rpki context exiting uses exit and not end

Issue:
The rpki subcontext uses exit instead of end to exit.
This makes issues with frr-reload in the way that frr-reload never exits
rpki context until it reaches the next end statement. this also happens when
parsing the configuration from vtysh.

Signed-off-by: Runar Borge <runar@borge.nu>
---
 tools/frr-reload.py | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/tools/frr-reload.py b/tools/frr-reload.py
index 412cde091..33c4ef5e5 100755
--- a/tools/frr-reload.py
+++ b/tools/frr-reload.py
@@ -452,6 +452,17 @@ end
                 ctx_keys = []
                 current_context_lines = []
 
+            elif (line == "exit" and
+                  ctx_keys[0].startswith('rpki')):
+                self.save_contexts(ctx_keys, current_context_lines)
+                log.debug('LINE %-50s: exiting old context, %-50s', line, ctx_keys)
+
+                # Start a new context
+                new_ctx = True
+                main_ctx_key = []
+                ctx_keys = []
+                current_context_lines = []
+
             elif line == "exit-vrf":
                 self.save_contexts(ctx_keys, current_context_lines)
                 current_context_lines.append(line)
-- 
2.20.1

